metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java

```diff
@@ -218,4 +223,39 @@ public SearchResult filter(
       int count,
       @Nonnull String actor)
       throws RemoteInvocationException;
+
+  public VersionedAspect getAspect(
+      @Nonnull String urn,
+      @Nonnull String aspect,
+      @Nonnull Long version,
+      @Nonnull String actor)
+      throws RemoteInvocationException;
+
+  public VersionedAspect getAspectOrNull(
+      @Nonnull String urn,
+      @Nonnull String aspect,
+      @Nonnull Long version,
+      @Nonnull String actor) throws RemoteInvocationException;
+
+  public List<EnvelopedAspect> getTimeseriesAspectValues(
+      @Nonnull String urn,
+      @Nonnull String entity,
+      @Nonnull String aspect,
+      @Nullable Long startTimeMillis,
+      @Nullable Long endTimeMillis,
+      @Nullable Integer limit,
+      @Nullable String actor
+  ) throws RemoteInvocationException;
+
+  public String ingestProposal(
+      @Nonnull final MetadataChangeProposal metadataChangeProposal,
+      @Nonnull final String actor
+  ) throws RemoteInvocationException;
+
+  public <T extends RecordTemplate> Optional<T> getVersionedAspect(
```

Comment:
QQ - Why is this "getVersionedAspect" when getAspect actually returns a versionedAspect and this just returns an aspect.. we should think about renaming or deprecating this one

Reply:
sg